hera/editor/behavior/DesignBehavior.tsx

```diff
@@ -228,11 +234,47 @@ export default class DesignBehavior {
     return null;
   }
 
+  private putMultiple(
+    vectors: Vector[],
+    state: State,
+    actions: Actions,
+    editor: EditorState,
+  ): StateLike | null {
+    let result: StateLike | null = null;
+    const players = Array.from(
+      new Set([...state.map.active, ...PlayerIDs]),
+    ).slice(0, vectors.length);
+    vectors.forEach((vector, index) => {
+      const currentPlayerIndex = players.indexOf(
+        state.map.getCurrentPlayer().id,
+      );
+      const playerId =
+        players[
+          ((currentPlayerIndex >= 0 ? currentPlayerIndex : 0) + index) %
+            players.length
+        ];
+
+      const tempResult = this.put(
+        vector,
+        { ...state, ...result },
+        actions,
+        editor,
+        playerId,
+      );
+      result = {
+        ...(result ?? {}),
+        ...tempResult,
+      };
```

Comment:
You shouldn't need the `?? {}` as you can spread null.

```suggestion
      const newState = this.put(
        vector,
        { ...state, ...result },
        actions,
        editor,
        playerId,
      );
      result = {
        ...(newState),
        ...tempResult,
      };
```

Reply:
I think `result` had a different type at some point. Removed the null check: e9eaa9a68f8ebc02571f0959de99716a502a2158

Some renaming: 387fe45b8702aea6803186dfef8dbc3e284fec10